server/auth/auth.js

```diff
@@ -113,6 +140,12 @@ exports.validateRequest = function(req, res, next) {
   );
 };
 
+/**
+ * Receives information regarding the user using promises
+ *
+ * @param {Object} req - Request received
+ * @return {Promise} returns the user data using promises
+*/
```

Comment:
This is the same as the function `getUserData`, but instead of having a response (`res`), we return a Promise using the [Q library](https://github.com/kriskowal/q) for a better handling of the user information